.github/workflows/tag.yml

```diff
@@ -0,0 +1,44 @@
+name: Tag
+
+on:
+  push:
+    tags: '*'
+
+jobs:
+  test:
+    name: Tag
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/setup-go@v3
+        with:
+          go-version: 1.18.7
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Style checks
+        run: |
+          make style
+          if ! git diff --exit-code HEAD; then
+            echo
+            echo "*** Files are not formatted properly. See the above diff for more info."
+            exit 1
+          fi
+      - name: Unit tests
+        run: |
+          make test
+      - name: Build
+        run: |
+          make build
+      - name: Dry run
+        run: |
+          make dry-run
+      - name: Build and Push Container Image
+        run: |
+          docker login -u "${{ secrets.QUAY_RHACS_ENG_RW_USERNAME }}" --password-stdin <<<"${{ secrets.QUAY_RHACS_ENG_RW_PASSWORD }}" quay.io
+          make push
+      - name: Re-tag and Push Latest
+        run: |
+          make tag-latest
+          make push-latest
```

Comment:
The only difference between `ci.yml` and `test.yml` that I spot is presence or absence of this step. Maybe let's keep a single file with shared trigger
```yaml
on:
  push:
    branches: [master]
    tags: '*'
  pull_request:
    types: [opened, synchronize]
```
and have a conditional logic inside the step that skips it when it's not a tagged build?
Something like this (untested):
```yaml
      - name: Re-tag and Push Latest
        run: |
          if [[ "${{ github.ref }}" =~ refs/tags/* ]]; then
            echo "This is not a tagged build. Skipping latest tag update."
            exit 0
          fi
          make tag-latest
          make push-latest
```
(see <https://docs.github.com/en/actions/learn-github-actions/contexts>)

Reply:
Works for me